src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -130,9 +187,41 @@ export default function withAnnotations<P extends object>(
             }
         };
 
+        handleAnnotationsFetchError = ({ error }: { error: SerializedError }): void => {
+            if (!error) {
+                return;
+            }
+
+            const { message = '' } = error;
+            const formattedError = {
+                code: ERROR_CODE_FETCH_ANNOTATIONS,
+                // eslint-disable-next-line @typescript-eslint/camelcase
+                context_info: error,
+                message,
+                origin: ORIGIN_CONTENT_PREVIEW,
+                type: 'error' as const,
+            };
+
+            this.setState({
+                error: formattedError,
+            });
+
+            this.escalateAnnotationsFetchError(formattedError);
+        };
+
         handleAnnotator = (annotator: Annotator): void => {
             this.annotator = annotator;
             this.annotator.addListener('annotatorevent', this.handleAnnotatorEvent);
+
+            this.setState({ hasAnnotator: true });
+        };
+
+        handleAnnotatorCreate = (annotator: Annotator): void => {
+            annotator.addListener('annotations_fetch_error', this.handleAnnotationsFetchError);
```

Comment:
It seems a bit strange to call `addListener` on the passed in `annotator` imo. Could `handleAnnotatorCreate` do the assignment to `this.annotator` and add the listeners for `annotations_fetch_error` and `annotatorevent` and then have `handleAnnotator` just set the state for `hasAnnotator`?

And along those lines, what do you think about changing the state from `hasAnnotator` => `hasAnnotatorInit`